mlflow/models/resources.py

```diff
@@ -58,6 +59,27 @@ class DatabricksResource(Resource, ABC):
     target_uri: str = "databricks"
 
 
+@dataclass
+class DatabricksUCConnection(DatabricksResource):
+    """
+    Define a Databricks UC Connection used to serve a model.
+
+    Args:
+        connection_name (str): The name of the databricks UC connection
+        used to create the tool which was used to build the model.
+    """
+
+    type: ResourceType = ResourceType.UC_CONNECTION
+    connection_name: str = None
```

Comment:
```suggestion
    connection_name: Optional[str] = None
```

Reply:
Indeed, what does it mean getting `None` here? Can connection name in UC be empty or does it have some sort of default name?

Reply:
I don't remember the exact reason, but there is some loop below which would filter out the {} to_dict as far as I remember. Its been a while we wrote this code, so don't remember.

Reply:
I don't want to add optional since it would add it to the doc string I think. So do you think we can still keep it as it is? What do you think?

Reply:
Hmm got it. Type annotation itself is not a big deal, but could you add some comment for the `None` reasoning?

Reply:
@sunishsheth2009 If you don't want us to add `Optional` here. Please add a comment. If I saw this, I would automatically add `Optional`.

Reply:
Yes makes sense. Since this is a minor version patch release change, let not make this change in this PR and will create another PR to update that. Merging this for now :) Will create a PR and link it here for ref